packages/ui/src/routes/hardware-wallet/RemoveDevice.tsx

```diff
@@ -71,48 +74,73 @@ const HardwareWalletRemoveDevicePage = () => {
-                        {isDeviceImported(Devices.LEDGER)} Accounts
+                        {importedLedger === 1
+                            ? "1 Account"
+                            : importedLedger + " accounts"}
                     </button>
                     <button
                         type="button"
                         onClick={() =>
-                            isDeviceImported(Devices.TREZOR) &&
-                            setSelectedVendor(Devices.TREZOR)
+                            importedTrezor && setSelectedVendor(Devices.TREZOR)
                         }
                         className={classnames(
                             "bg-white rounded-md justify-center p-4 w-1/2 flex flex-col items-center group space-y-3 cursor-pointer border hover:border-primary-300",
                             selectedVendor === Devices.TREZOR
                                 ? "border-primary-300"
                                 : "border-primary-100",
-                            !isDeviceImported(Devices.TREZOR) &&
-                                "opacity-50 pointer-events-none"
+                            !importedTrezor && "opacity-50 pointer-events-none"
                         )}
                         style={{ height: "120px" }}
                     >
                         <img
                             src={trezor}
-                            alt="Connect Trezor"
-                            className="h-8"
+                            alt="Remove Trezor hardware wallet"
+                            className="h-8 mb-2"
+                        />
+                        {importedTrezor === 1
+                            ? "1 Account"
+                            : importedTrezor + " accounts"}
+                    </button>
+                    <button
+                        type="button"
+                        onClick={() =>
+                            importedKeystone &&
+                            setSelectedVendor(Devices.KEYSTONE)
+                        }
+                        className={classnames(
+                            "bg-white rounded-md justify-center p-4 w-1/2 flex flex-col items-center group space-y-3 cursor-pointer border hover:border-primary-300",
+                            selectedVendor === Devices.KEYSTONE
+                                ? "border-primary-300"
+                                : "border-primary-100",
+                            !importedKeystone &&
+                                "opacity-50 pointer-events-none"
+                        )}
+                        style={{ height: "120px" }}
+                    >
+                        <img
+                            src={keystone}
+                            alt="Remove Keystone hardware wallet"
+                            className="h-8 mb-2"
                         />
-                        {isDeviceImported(Devices.TREZOR)} Accounts
+                        {importedKeystone === 1
```

Comment:
Same as above. Can you extract this code so that we don't have to repeat it?

Reply:
I saw we're using it for every device type

Reply:
Also, the button itself cannot be extracted so that we don't have to repeat it 3 times?